File: src/api/LMSEntities/DataTransferObjects/LibraryAssetForUpdateDto.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace LMSEntities.DataTransferObjects
{
    public class LibraryAssetForUpdateDto
    {
        [Required]
        public int Id { get; set; }
        [Required]
        public string Title { get; set; }
        [Required]
        public int Year { get; set; }
        [Required]
        public int NumberOfCopies { get; set; }

        [Required]
        public int CopiesAvailable { get; set; }

        [Required]
        public string Description { get; set; }
        [Required]
        public LibraryAssetTypeDto AssetType { get; set; }
        public string ISBN { get; set; }
        public string DeweyIndex { get; set; }

        // [Required]
        public ICollection<LibraryAssetAuthorDto> AssetAuthors { get; set; }

        // [Required]
        public ICollection<LibraryAssetCategoryDto> AssetCategories { get; set; }
    }
}

File: src/spa/src/dto/models/library-asset-for-update-dto.ts
/* tslint:disable */
/* eslint-disable */
import { LibraryAssetAuthorDto } from './library-asset-author-dto';
import { LibraryAssetCategoryDto } from './library-asset-category-dto';
import { LibraryAssetTypeDto } from './library-asset-type-dto';
export interface LibraryAssetForUpdateDto {
  assetAuthors: Array<LibraryAssetAuthorDto>;
  assetCategories: Array<LibraryAssetCategoryDto>;
  assetType: LibraryAssetTypeDto;
  copiesAvailable: number;
  description: string;
  deweyIndex?: null | string;
  id: number;
  isbn?: null | string;
  numberOfCopies: number;
  title: string;
  year: number;
}
